exp/addons/internal/controllers/clusterresourcesetbinding_controller.go

```diff
@@ -77,8 +78,33 @@ func (r *ClusterResourceSetBindingReconciler) Reconcile(ctx context.Context, req
 		// Error reading the object - requeue the request.
 		return ctrl.Result{}, err
 	}
-
-	cluster, err := util.GetOwnerCluster(ctx, r.Client, binding.ObjectMeta)
+	// Before 1.4 cluster name was stored as ownerRef; this code migrated this value to
+	// the spec.clusterName field if necessary.
+	var clusterName = binding.Spec.ClusterName
+	if clusterName == "" {
```

Comment:
https://github.com/kubernetes-sigs/cluster-api/pull/7680/files#r1060785201

If `clusterName` is set as a pointer, then not only the case where `clusterName` is `nil` but also the case where `*clusterName` is `""` is checked here, right?

@killianmuldoon

Reply:
I think it's sufficient to check that it's nil here. This should never be changed by another controller.